FNZ-TimeTracking-1- Added User entities

Wejdene-CH opened the pull request.

Wejdene-CH committed:
FNZ-TimeTracking-1- Added User entities

baccartaoufik commented on src/main/java/com/fnz/TimeTracking/model/Utilisateur.java:
```diff
@@ -0,0 +1,45 @@
+    private Set<Départ> departs;
+
+    @OneToMany(mappedBy = "utilisateur")
+    private Set<Permission> permissions;
```
The user in total has 3 DIRECT relations:
1. User > Pointage : OneToMany
2. User > Sentiments: OneToMany
3. User > Congé: OneToMany

Also notice that your mappedBy field is set to 'utilisateur' in every relation, that's wrong and must be changed

baccartaoufik commented on src/main/java/com/fnz/TimeTracking/model/Pointage.java:
```diff
@@ -0,0 +1,24 @@
+    private LocalDateTime clockIn;
+    private LocalDateTime clockOut;
+
+
```
In the provided class diagram, this class should have 3 direct relations:

Pointage > Utilisateur: ManyToOne
Pointage > Depart: OneToOne
Pointage > Arrivée: OneToOne

baccartaoufik commented on src/main/java/com/fnz/TimeTracking/model/Permission.java:
```diff
@@ -0,0 +1,22 @@
+    private LocalDateTime heureSortie;
+    private LocalDateTime heureRetour;
+
+    @ManyToOne
```
This relation is wrong because the User has direct connection with Conge, so this must be removed.

instead, add A OneToOne relation with Permission > Conge

baccartaoufik commented:
Overall the code looks great and well ogranized.
The project structure also corresponds to the proposed one.
Just few missing relations that need to be added.

baccartaoufik reviewed (changes requested):
Overall the code looks great and well ogranized.
The project structure also corresponds to the proposed one.
Just few missing relations that need to be added.

Wejdene-CH committed:
FNZ-TimeTracking-1-fix: Add missing relations in entity classes

- Added missing relations between entities to ensure proper data modeling.
- Verified and corrected connections between Utilisateur, Pointage, Départ,arrivée,Sentiment and Role entities.
- Ensured all necessary annotations and mappings are in place.

Commit:
FNZ-TimeTracking-1 Create database if not exists
